### ISSUE TITLE
Suggestion: Sign the Eclipse packages

### ISSUE THREAD

**Issue by hrj** [vrapper/vrapper#593]
Whenever I update the plugins, I get an unsigned package warning. (vrapper is not alone in this; even Android Developer Tools are unsigned).

Would it be possible to sign them? I am not sure how securely Eclipse verifies the signatures but, hopefully, it will prevent MITM attacks.

**Comment by loonies** [comment]
:+1: I would like it as well.

**Comment by keforbes** [comment]
I looked into this a little bit and I'm not sure if there's a legitimate solution.  It seems the most common answer is to create a self-signed certificate.  This means when you update the package Eclipse asks "do you trust this certificate?"  I know this happens with PyDev.  I don't know how to permanently trust a certificate so I always have to accept that dialog whenever I update PyDev.

It seems to me we either have an unsigned jar and you have to say "yes" to "do you want to install these unsigned packages" or we create a self-signed certificate and you have to say "yes" to "do you trust this certificate".  Neither solution seems particularly secure or more usable to the end user.

Can I ask why you guys think this is important?  Is someone out there executing complex MITM attacks to serve up a fake Vrapper install?

**Comment by hrj** [comment]
> It seems to me we either have an unsigned jar and you have to say "yes" to "do you want to install these unsigned packages" or we create a self-signed certificate and you have to say "yes" to "do you trust this certificate". Neither solution seems particularly secure or more usable to the end user.

Yeah a self-signed certificate will not be much better, especially if the IDE is not going to remember it.

However, I think there are CAs that, for free, give a signing certificate to open-source projects. If willing to spend some cash, then there are many more options. I need to do something like this for my own open-source projects too.. but haven't done the complete research yet (waiting for them to reach this level of popularity).

Thinking more about this: if signing is going to be too much of a hassle, perhaps the best way is for developers to compile it locally by themselves. It is feasible for this project, since it is targeted at developers. In this regard, it would be nice if vrapper (or any other Eclipse plugin) could be compiled with a simple command-line ant / maven / gradle command.

> Can I ask why you guys think this is important? Is someone out there executing complex MITM attacks to serve up a fake Vrapper install?

I would wager that by now there are hundreds of agencies around the world, who ask such attacks to be done ... during interviews for an entry level hacker position (with a side-note to not use ready-made scripts). IMO, `complex` is not an appropriate adjective for it any more. `Improbable` perhaps, depending on which hat you wear :wink:.

**Comment by albertdev** [comment]
> Thinking more about this: if signing is going to be too much of a hassle, perhaps the best way is for developers to compile it locally by themselves. It is feasible for this project, since it is targeted at developers. In this regard, it would be nice if vrapper (or any other Eclipse plugin) could be compiled with a simple command-line ant / maven / gradle command.

I wouldn't recommend it, building Eclipse plugins is notoriously difficult without some infrastructure to host the dependencies (and the standard Maven repo doesn't host too much Eclipse stuff).

If anyone wants to build Vrapper from source, so be it. I think most users still prefer to pull in an unsigned build instead of downloading source code and installing some tools to build it themselves (even if [gradle wrapper](https://gradle.org/docs/current/userguide/gradle_wrapper.html) can make that process somewhat easier).

**Comment by loonies** [comment]
Is it possible to build and sign package for manually installation? We
download packages, verify signatures, and manually drop them into
directory(es), or do whatever the installation procedure is.
